conbench/entities/benchmark_result.py

```diff
@@ -363,15 +361,18 @@ def is_failed(self):
         The criteria are conventions that we (hopefully) apply consistently
         across components.
         """
-        if self.data is None:
+        if self.unit is None:
             return True
 
-        if do_iteration_samples_look_like_error(self.data):
+        if self.data is None:
             return True
 
         if self.error is not None:
             return True
 
+        if do_iteration_samples_look_like_error(self.data):
+            return True
+
```

Comment:
changes in this section: least expensive checks first, and now also use absence of unit as criterion for "failed result".

Reply:
lovely!